### PR TITLE
Unit tests for Gobblin Yarn part II

### PR DESCRIPTION
**Pull request by liyinan926**
This is the part II of unit tests for Gobblin Yarn. Part II additionally covers `YarnAppSecurityManager` and `YarnContainerSecurityManager`. Currently the only Yarn-dependent class that has no coverage is `YarnService`. Since `YarnService` runs in the `GobblinApplicationMaster` that is supposed to be launched by Yarn without user interference, I'm still trying to figure out a way of running it manually. This PR also addresses the left-over review comments from PR #501.  

Signed-off-by: Yinan Li <liyinan926@gmail.com>

### CODE REVIEW

**Comment by liyinan926** [comment]
@sahilTakiar can you review?

File: gobblin-yarn/src/test/java/gobblin/yarn/YarnSecurityManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    this.helixManager = HelixManagerFactory.getZKHelixManager(
+        helixClusterName, TestHelper.TEST_HELIX_INSTANCE_NAME, InstanceType.SPECTATOR, zkConnectingString);
+    this.helixManager.connect();
+
+    this.miniHDFSBaseDir = new Path(YarnSecurityManagerTest.class.getSimpleName());
+    this.dfsConfig = getTestConfiguration(this.miniHDFSBaseDir);
+    dfsConfig.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, this.miniHDFSBaseDir.toUri().toString());
+    MiniDFSCluster.Builder builder = new MiniDFSCluster.Builder(dfsConfig);
+    this.miniDFSCluster = builder.numDataNodes(1).build();
```
Comment (sahilTakiar) [inline gobblin-yarn/src/test/java/gobblin/yarn/YarnSecurityManagerTest.java]: You can combine the above two lines

**Comment by sahilTakiar** [comment]
Do we have any documentation on the testing strategy?

**Comment by liyinan926** [comment]
What you mean by testing strategy?

**Comment by sahilTakiar** [comment]
More like what the tests do, how we are using the curator framework, how we are using the MiniDfsCluster, etc. 

* My understanding is that we are using the Curator framework to setup a local Zookeeper instance that Helix connects to, is that correct?
* Why is it necessary to use the Curator framework vs. just mocking HelixManager
* What is MiniDfsCluster needed for? Why is it necessary to use a MiniDfsCluster over a LocalFileSystem.
* Are these all unit tests? Is it possible to have an end-to-end integration tests.

**Comment by liyinan926** [comment]
1. The Curator framework is a ZooKeeper client that is easier to use than the official ZooKeeper client library. 
2. The `MiniDFSCluster` was meant to provide a way to get a delegation token. I just realized it's not really necessary since it's essentially the same as `LocalFileSystem` for this, both return `null`, and that's why I need to use mocking. 
3. Yes, these are unit tests. 
4. Yes, I agree we need end-to-end integration tests.

**Comment by liyinan926** [comment]
@sahilTakiar updated 29f4e46.

**Comment by liyinan926** [comment]
Hey, @sahilTakiar, can you take another look d11cd1f?

File: gobblin-utility/src/main/java/gobblin/util/ParallelRunner.java [hunk 2]
```diff
@@ -246,7 +246,7 @@ public void close() throws IOException {
     } catch (InterruptedException ie) {
       throw new IOException(ie);
     } catch (ExecutionException ee) {
-      throw new IOException(ee);
+      throw new IOException(ee.getCause());
```
Comment (sahilTakiar) [inline gobblin-utility/src/main/java/gobblin/util/ParallelRunner.java]: Why make this change? I believe this won't preserve the stack trace of the original exception.

Comment (liyinan926) [inline gobblin-utility/src/main/java/gobblin/util/ParallelRunner.java]: `ExecutionException` is just a wrapper of the actual exception returned by `getCause`.

**Comment by sahilTakiar** [comment]
LGTM

Only comment is why do we need the Curator Framework when we could just mock the `HelixManager` using Mockito? Would making a mock `HelixManager` be hard? The advantage of just using a mock is that we don't have to do things like `Thread.sleep(5000)` in the code to wait for the Helix + ZK interaction to complete.

**Comment by liyinan926** [comment]
A `HelixManager` does not allow you to check arbitrary ZooKeeper path. In those tests, I really want to make sure the appropriate ZooKeeper paths get created in real. A mocked `HelixManager` simply won't do the job.

**Pull request merged**